### PR TITLE
refactor(i18n): to load the i18n messages async

### PR DESCRIPTION
**Pull request by emmenko**
Following the [suggestions from ui-kit](https://github.com/commercetools/ui-kit/pull/643), we can actually treat the ui-kit translations as we do for the other localization files.

### CODE REVIEW

File: packages/application-shell/src/components/application-shell-provider/application-shell-provider.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import './global-style-imports';
```
Comment (montezume) [inline packages/application-shell/src/components/application-shell-provider/application-shell-provider.js]: makes sense to put this at the top 👍

File: packages/i18n/load-i18n.js [hunk 2]
```diff
@@ -38,7 +39,23 @@ const getMomentChunkImport = locale => {
   }
 };
 
-const getLocalizedStringsChunkImport = locale => {
+const getUiKitChunkImport = locale => {
+  const intlLocale = mapLocaleToIntlLocale(locale);
+  switch (intlLocale) {
+    case 'de':
+      return import(/* webpackChunkName: "i18n-ui-kit-locale-de" */ '@commercetools-frontend/ui-kit/i18n/de.json');
```
Comment (emmenko) [inline packages/i18n/load-i18n.js]: FYI: we just need to remember to update the path to `.../i18n/data/...` once we do the next ui-kit release

Comment (tdeekens) [reply packages/i18n/load-i18n.js]: Yeah it's funny that these things tend to be breaking releases from the view of UIKit.

Comment (montezume) [reply packages/i18n/load-i18n.js]: seems like an easy thing to forget 🙈

Comment (emmenko) [reply packages/i18n/load-i18n.js]: It depends when we are going to release ui-kit :)

We can also revert the folder structure in the ui-kit...

Comment (emmenko) [reply packages/i18n/load-i18n.js]: I guess we need to wait anyway for the ui-kit release, as we expect the new locales to be there

**Comment by emmenko** [comment]
Waiting for a new ui-kit release, in the meantime this is blocked.

**Comment by montezume** [comment]
Can use ui-kit `9.2.0`

**Comment by emmenko** [comment]
FYI: there are some failing VRT tests (unrelated to this PR). There is a related PR to fix one of the problems (#542)

**Pull request merged**